### PR TITLE
Add Contributors guidance for Dioptra

### LINKED ISSUE #80: Write Contributors guidance for Dioptra
## Definition of Done

- [x] A CONTRIBUTORS.md file is available in the repo
- [x] Describes Issue Templates
- [x] Development and Pull Request Workflow
- [x] Commit Styleguide
- [x] Contributor's license

### LINKED ISSUE #81: Write a Code Review Guide
## Definition of Done

- [x] Outlines the governance for reviewing code provided in a pull request (number of reviews needed for acceptance, who's authorized to approve the final merge, separation of submitter and approval roles)
- [x] Governance for squashing/combining commit messages
- [x] A file CODEREVIEWERS.md exists in the repo
- [x] This document will include suggestions on things to look for during a code review and guidance on how to provide constructive and actionable feedback

### PR DESCRIPTION
**Pull request by jkglasbrenner**
Closes #80 
Closes #81

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+# Submitting Issues
+
+Prior to submitting any issue please try a brief search to see if an issue similar to yours already exists. Avoiding duplicate submissions helps to limit time needed to triage issues. Each issue card should be focused and describe a single bug, feature, etc… Please try to use the [provided templates](https://github.com/usnistgov/dioptra/issues/new/choose) if they align with the issue you are submitting.
+
+## Bugs
+
+If you find a bug or unexpected behavior please use the provided [issue template](https://github.com/usnistgov/dioptra/issues/new/choose) and describe the issue along with step-by-step instructions on how to reproduce the issue.
+
+# Feature Requests
```
Comment (hbooth) [inline CONTRIBUTING.md:49]: Perhaps be a sub-header (in other words needs another '#' character)?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+Create a [new branch](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging) to get started.
+
+### Squashing
+
+All final commits will be squashed, therefore when squashing your branch, it’s important to make sure you update the commit message. If you’re using Github’s UI it will by default create a new commit message which is a combination of all commits and **does not follow the commit guidelines**.
+
+If you’re working locally, it often can be useful to `--amend` a commit, or utilize `rebase -i` to reorder, squash, and reword your commits.
+
+# Commit Style Guide
```
Comment (hbooth) [inline CONTRIBUTING.md:75]: I think may also be a sub-heading too. Either that, or the next one should be a top-level header, although I think sub-heading may make more sense.

**Review by hbooth — CHANGES_REQUESTED** [review]
Just the two comments and it should be ready to go.

**Comment by jkglasbrenner** [comment]
@hbooth I looked at the Markdown headers you pointed out and agree they were at the wrong level. After looking at CONTRIBUTING.md again in light of your feedback, here's the new subject heading hierarchy:

- Criteria for Contributions and Feedback
- Contributor Responsibilities
- Question or Problem?
- Submitting Issues
    - Bugs
    - Feature Requests
- Code Submission Guidelines
    - Development Workflow
    - Commit Style Guide
        - Squashing
    - Code Reviewing Guide

I moved the squashing section under Commit Style Guide since that seemed to make more sense, and kept Code Reviewing Guide as a sub-heading up Code Submission Guidelines.

Also, I didn't notice that we had more than one `#` header in the file, the convention is to reserve `#` for the title and only use it once. I also fixed that up.

I think it looks good now, let me know if you agree and we can get this merged.

**Review by hbooth — APPROVED** [review]
Agreed and everything looks good to me.

**Pull request merged**